service/s3/unmarshal_error.go

```diff
@@ -48,6 +53,7 @@ func unmarshalError(r *request.Request) {
 	} else {
 		errCode = resp.Code
 		errMsg = resp.Message
+		err = nil
```

Comment:
Do we want to squash a potential io.EOF? I feel this could be meaningful information.